packages/fiori/src/BarcodeScannerDialog.js

```diff
@@ -0,0 +1,250 @@
+import UI5Element from "@ui5/webcomponents-base/dist/UI5Element.js";
+import litRender from "@ui5/webcomponents-base/dist/renderer/LitRenderer.js";
+import { getI18nBundle } from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import Dialog from "@ui5/webcomponents/dist/Dialog.js";
+import Button from "@ui5/webcomponents/dist/Button.js";
+import BusyIndicator from "@ui5/webcomponents/dist/BusyIndicator.js";
+import { BrowserMultiFormatReader, NotFoundException } from "@zxing/library";
+
+// Template
+import BarcodeScannerDialogTemplate from "./generated/templates/BarcodeScannerDialogTemplate.lit.js";
+
+// Styles
+import barcodeScannerDialogCss from "./generated/themes/BarcodeScannerDialog.css.js";
+
+// Texts
+import {
+	BARCODE_SCANNER_DIALOG_CANCEL_BUTTON_TXT,
+	BARCODE_SCANNER_DIALOG_LOADING_TXT,
+} from "./generated/i18n/i18n-defaults.js";
+
+const defaultMediaConstraints = {
+	audio: false,
+	video: {
+		height: {
+			min: 480,
+			ideal: 960,
+			max: 1440,
+		},
+		aspectRatio: 1.333333333,
+		facingMode: "environment",
+	},
+};
+
+/**
+ * @public
+ */
+const metadata = {
+	tag: "ui5-barcode-scanner-dialog",
+	languageAware: true,
+	slots: /** @lends  sap.ui.webcomponents.fiori.BarcodeScannerDialog.prototype */ {
+	},
+	properties: /** @lends  sap.ui.webcomponents.fiori.BarcodeScannerDialog.prototype */ {
+		/**
+		 * Indicates whether a loading indicator should be shown in the dialog.
+		 *
+		 * @type {boolean}
+		 * @defaultvalue false
+		 * @private
+		 */
+		loading: {
+			type: Boolean,
+		},
+	},
+	events: /** @lends  sap.ui.webcomponents.fiori.BarcodeScannerDialog.prototype */ {
+		/**
+		 * Fired when the scan completed successfuuly.
+		 *
+		 * @event sap.ui.webcomponents.fiori.BarcodeScannerDialog#scan-success
+		 * @param {String} result the scan result.
+		 * @public
+		 */
+		 "scan-success": {
+			detail: {
+				text: { type: String },
+				rawBytes: { type: Object },
+			},
+		},
+
+		/**
+		 * Fired when the scan failed with error.
+		 *
+		 * @event sap.ui.webcomponents.fiori.BarcodeScannerDialog#scan-error
+		 * @param {String} message the error message.
+		 * @public
+		 */
+		 "scan-error": {
+			detail: {
+				message: { type: String },
+			},
+		},
+	},
+};
+
+/**
+ * @class
+ *
+ * <h3 class="comment-api-title">Overview</h3>
+ *
+ * Opening the dialog launches the device camera and scans for known barcode formats.
```

Comment:
Maybe adding one additional intro sentence before "Opening the dialog launches the device camera..." or perhaps you can check it with the team's UA.

Reply:
Done. Will send to UA as well.
(btw Is there a UA assigned for the ui5-web-components in particular)?
Thnx